### ISSUE TITLE
cabal-to-dhall golden tests requires diff program (not present by default in win)

### ISSUE THREAD

**Issue by jneira** [dhall-lang/dhall-to-cabal#72]
Hi! Afaik the awesome new test group `cabal-to-dhall` by @quasicomputational needs to execute the diff program and the poor souls that are in windows doesn't have it by default, ¿could be easily replaced with calls to Diff haskell package (already in scope)?

In windows all the tests fail with
```
  cabal-to-dhall
    simple:                   FAIL
      Exception: diff: createProcess: does not exist (No such file or directory)

    gh-36:                    FAIL
      Exception: diff: createProcess: does not exist (No such file or directory)

    conditional-dependencies: FAIL
      Exception: diff: createProcess: does not exist (No such file or directory)
```

There are workarounds:
* download diff for windows: http://gnuwin32.sourceforge.net/packages/diffutils.htm
* use a mingw bash console (?)
  * [changing the character encoding to utf-8](https://stackoverflow.com/a/36692549/49554)
  * get cabal/ghc to work with utf-8 when executed in mingw bash (i've not be able to do it yet, change all locales to C.UTF-8 doest not work for me)

(Tested in windows 7 with 7fe5edd version and cabal 2.2.0.0)

**Comment by quasicomputational** [comment]
Huh, fair enough. The path of least resistance through `tasty-golden`'s API is to use an external diff and that's the way I went, yeah. We've already got an example of a pure-Haskell diff in the test suite (added by @jneira himself!), so fixing this shouldn't be too hard if there's a decent Haskell implementation of text diffing anywhere.

**Comment by jneira** [comment]
@quasicomputational as there is a relativily easy workaround (and to have diff is valuable in its own!), i think it is not a priority. I'll try to change it when i have time

**Comment by ocharles** [comment]
Fixed by #90
